### PR TITLE
Add remaining features to "What's New in 3.7" reference documentation

### PR DESCRIPTION
**Pull request by jyemin**
Published the docs to my fork: http://jyemin.github.io/mongo-java-driver/3.7/whats-new/

They should arrive there shortly.

### CODE REVIEW

File: docs/reference/content/whats-new.md [hunk 1]
```diff
@@ -63,6 +63,48 @@ Support for `Instant`, `LocalDate` and `LocalDateTime` has been added to the dri
 and / or more flexibility regarding JSR-310 dates should check out the alternative JSR-310 codecs provider by Cezary:
 [bson-codecs-jsr310](https://github.com/cbartosiak/bson-codecs-jsr310).
 
+### JSR-305 NonNull annotations
+
+The public API is now annotated with JSR-305 compatible @NonNull and @Nullable annotations.  This will allow programmers
```
Comment (rozza) [inline docs/reference/content/whats-new.md]: nit: backtick the annotations eg: `@NonNull`

File: docs/reference/content/whats-new.md [hunk 2]
```diff
@@ -63,6 +63,48 @@ Support for `Instant`, `LocalDate` and `LocalDateTime` has been added to the dri
...
+to rely on tools like FindBugs/SpotBugs, IDEs like IntelliJ IDEA, and compilers like the Kotlin compiler to find errors in the use of the 
+driver via static analysis rather than via runtime failures, which may happen infrequently.
```
Comment (rozza) [inline docs/reference/content/whats-new.md]: This is hard to parse - suggest removing the last part ', which may happen infrequently.'

File: docs/reference/content/whats-new.md (hunk 1 above)
Comment (jyemin) [reply docs/reference/content/whats-new.md]: Done

File: docs/reference/content/whats-new.md (hunk 2 above)
Comment (jyemin) [reply docs/reference/content/whats-new.md]: Done

**Comment by jyemin** [comment]
Squashed and merged.
